float.s

```diff
@@ -74,15 +96,15 @@ ___float_u16_to_fac:
 __float_u16_to_fac:
         sta FAC_MANTISSA0
         sty FAC_MANTISSA1
-        jsr __basicon
+        __enable_basic_if_needed
         ldx #$90
         sec
         jsr BASIC_u16_to_FAC
-        jmp __basicoff
+        __return_with_cleanup
 
 ; return to C, FAC as unsigned int
 __float_fac_to_u16:
-        jsr __basicon
+        __enable_basic_if_needed
         jsr BASIC_FAC_to_u16
         jsr __basicoff
```

Comment:
please also substitute the __basicoff calls by a macro (__disable_basic_if_needed)

Reply:
OK, I think I originally misread this. I already have a macro `__return_with_cleanup` that I should have used here. I gave it an "asymmetric" because `__disable_basic_if_needed` doesn't make it clear that the macro results in a change to execution flow.